### DISCUSSION TITLE
NUCLEO WB55RG : debugging with openocd when using USB CDC

### DISCUSSION THREAD

**Discussion by jmchiappa** [stm32duino/Arduino_Core_STM32#2327] [Variants] [answered]
Hello,

I'm usually using openocd for debug purpose. For some reasons, I need to use the USB output in CDC mode so that all serial data will be streamed through it.
Unfortunately, all openocd debug failed with init error. I digged in the startup code and it seems that, when USB is enabled, some pins are initialized from the variant file `variants\STM32WBxx\WB55R(C-E-G)V\PeripheralPins.c`
```
//*** USB ***

#if defined(HAL_PCD_MODULE_ENABLED) || defined(HAL_HCD_MODULE_ENABLED)
WEAK const PinMap PinMap_USB[] = {
  {PA_11, USB, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_NOPULL, GPIO_AF10_USB)}, // USB_DM
  {PA_12, USB, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_NOPULL, GPIO_AF10_USB)}, // USB_DP
  {PA_13, USB, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF10_USB)}, // USB_NOE
  {NC,    NP,  0}
};
```
I understand the 2 first lines DP and DM. But PA13 is SWDIO and reinitializing it will always fail the debug session.

Is it necessary to reinit this pin ? I've commented it and my code works like a charm...

```
framework : STM32Duino 2.7.1
IDE ; Arduino IDE 2.3.2
board : PNUCLEO STM32WB55RG
```

**Comment by fpistm** [comment, marked as answer]
Right. You can simply redefine the array at sketch level as it is weak.
